pydoctor/test/test_epydoc2stan.py

```diff
@@ -856,3 +856,77 @@ def test_annotation_formatter(annotation: str) -> None:
     assert html.endswith('</code>')
     text = html[6:-7]
     assert text.replace('<wbr></wbr>', '').replace('<wbr>\n</wbr>', '') == expected_text
+
+def test_module_docformat(capsys: CapSys) -> None:
+    """
+    Test if Module.docformat effectively override System.options.docformat
+    """
+
+    system = model.System()
+    system.options.docformat = 'restructuredtext'
+
+    mod = fromText('''
+    """
+    Link to pydoctor: U{pydoctor <https://github.com/twisted/pydoctor>}.
+    """
+    __docformat__ = "epytext"
+    ''', modname='test_epy', system=system)
+
+    epytext_output = epydoc2stan.format_docstring(mod)
+
+    captured = capsys.readouterr().out
+    assert not captured
+
+    mod = fromText('''
+    """
+    Link to pydoctor: `pydoctor <https://github.com/twisted/pydoctor>`_.
+    """
+    __docformat__ = "restructuredtext en"
+    ''', modname='test_rst', system=system)
+
+    restructuredtext_output = epydoc2stan.format_docstring(mod)
+
+    captured = capsys.readouterr().out
+    assert not captured
+
+    assert ('Link to pydoctor: <a href="https://github.com/twisted/pydoctor"'
+        ' target="_top">pydoctor</a>' in flatten(epytext_output))
+    
+    assert ('Link to pydoctor: <a class="rst-reference external"'
+        ' href="https://github.com/twisted/pydoctor" target="_top">pydoctor</a>' in flatten(restructuredtext_output))
+
```

Comment:
I think that what comes after this point should be a separate test case.

For the test case above, perhaps it can be shortened by making the docstring a parameter to the test function and unifying the other code by checking only for `href="https://github.com/twisted/pydoctor"`. The latter might be a good idea anyway to avoid having to update this test case when formatting details change.

Note that for `"restructuredtext en"` you're actually not testing whether the override works, but only whether the `en` part doesn't trigger any warnings. Maybe the `system.options.docformat` could be set to `plaintext` to test the override mechanism for both docstrings.

Reply:
Thanks! I've applied those comments